pytorch_lightning/trainer/connectors/accelerator_connector.py

```diff
@@ -703,6 +710,8 @@ def select_strategy(self) -> Strategy:
             plugin = DeepSpeedStrategy(
                 cluster_environment=self.select_cluster_environment(), parallel_devices=self.parallel_devices
             )
+        elif self.use_ddp and self.use_bagua:
```

Comment:
I followed the pattern with `DeepSpeedStrategy`, it seems`DeepSpeedStrategy` and `BaguaStrategy` both inherit from `DDPStrategy`.

Reply:
I think just `use_bagua` should work fine because `self._strategy_type` can have only one value

Reply:
but within lightning it will still fall under `self._strategy_type='bagua'` and user should get an error. But nvm, I think it will be resolved soon later after an ongoing refactor.